frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -15,7 +15,10 @@ export const LocaleProvider = ({
     if (locale) {
       setIsLoadingLocale(true);
       setLocaleHeader(locale);
-      loadLocalization(locale).then(() => setIsLoadingLocale(false));
+      loadLocalization(locale).then(translatedObject => {
+        setIsLoadingLocale(false);
+        setUserLocale(translatedObject);
```

Comment:
SDK doesn't need this fix, since there's no `window.MetabaseUserLocalization` or `window.MetabaseSiteLocalization`, so no override would ever happen.

https://github.com/metabase/metabase/blob/88b03e22cd483b9fc6e0b191539246d9d6cedfeb/frontend/src/metabase/lib/i18n.js#L165-L173